### ISSUE TITLE
Feature request: per-source `check-sourced` directive

### ISSUE THREAD

**Issue by ShamrockLee** [koalaman/shellcheck#2956]
#### For bugs
- Rule Id (if any, e.g. SC1000): 
- My shellcheck version (`shellcheck --version` or "online"): 
- [ ] The rule's wiki page does not already cover this (e.g. https://shellcheck.net/wiki/SC2086)
- [ ] I tried on https://www.shellcheck.net/ and verified that this is still a problem on the latest commit

#### For new checks and feature suggestions
- [ ] https://www.shellcheck.net/ (i.e. the latest commit) currently gives no useful warnings about this
- [X] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```bash
#!/usr/bin/env bash

# I don't want to check this one.
# shellcheck check-sourced=false
source messy.sh 

# I would like to have this checked.
# shellcheck check-sourced=true
source clean.sh

# The rest of the script goes here
echo "Hello, world!"
```

#### Here's what shellcheck currently says:
ShellCheck currently doesn't provide such directive.

```

In temp.sh line 4:
# shellcheck check-sourced=false
             ^-- SC1107 (warning): This directive is unknown. It will be ignored

In temp.sh line 8:
# shellcheck check-sourced=true
             ^-- SC1107 (warning): This directive is unknown. It will be ignored
```

#### Here's what I wanted or expected to see:

ShellCheck includes warning from `clean.sh` but not from `messy.sh`.

#### Motivation:

1. Chances are there that developers want to include warnings from some sourced files but not from others. This is especially valid if `messy.sh` cannot be changed from the project scope.
2. This feature provide a workaround for #356, by allowing developers to define a meta-script that `source`s the pre-sourced files before sourcing the target script to check. The directive proposed here helps exclude warnings from pre-sourced files, while include warnings from the target script.

**Comment by ConnorBaker** [comment]
While not satisfactory, I have found this at least useful while working on Nixpkgs:

```bash
# shellcheck source=<relative path to pkgs/stdenv/generic/setup.sh>
source /dev/null
```

**Comment by ShamrockLee** [comment]
> While not satisfactory, I have found this at least useful while working on Nixpkgs:
> 
> ```shell
> # shellcheck source=<relative path to pkgs/stdenv/generic/setup.sh>
> source /dev/null
> ```

What an excellent workaround for #356!

Per-source check-sourced would still be helpful when refactoring a codebase filled with un-lint scripts or excluding the warnings from an external shell library.
